engine/src/main/java/org/kigalisim/engine/recalc/SalesRecalcStrategy.java

```diff
@@ -68,20 +69,36 @@ public void execute(Engine target, RecalcKit kit) {
     EngineNumber initialChargeRaw = target.getInitialCharge("sales");
     EngineNumber initialCharge = unitConverter.convert(initialChargeRaw, "kg / unit");
 
-    // Get recycling volume
-    stateGetter.setVolume(rechargeVolume);
-    EngineNumber recoveryVolumeRaw = streamKeeper.getRecoveryRate(scopeEffective);
-    EngineNumber recoveryVolume = unitConverter.convert(recoveryVolumeRaw, "kg");
-    stateGetter.clearVolume();
+    // Get recycling volume and amount for both EOL and recharge stages
+    BigDecimal totalRecycledKg = BigDecimal.ZERO;
```

Comment:
Instead of starting this at zero and adding to it, let's just do the addition down below and declare the variable there.